### PR TITLE
Generates comprehensive Artwork type

### PR DESCRIPTION
**Pull request by dzucconi**
So this is still *experimental* — but here's a simple naive script to generate the Artwork type from scratch. This just simply iterates over chunks of 100 artworks until we run out of pages and utilizes the code-generation I built out in [this package](https://github.com/dzucconi/white-cube) to reduce it down. I believe the resulting type is a reasonably trustworthy representation that can actually be used.

One of the things that I think should be done is to either optionally treat literal `""` as possible `nulls` or to just include them as `R.Literal("")` because of how they tend to be used in JSX.

<details>
  <summary>📦 Published PR as canary version: <code>0.0.4-canary.4.21.0</code></summary>
  <br />
  
  :sparkles: Test out this PR locally via:
  
  ```sh
  npm install @artsy/quartz@0.0.4-canary.4.21.0
  # or 
  yarn add @artsy/quartz@0.0.4-canary.4.21.0
  ```
</details>

### CODE REVIEW

File: src/gravity/Artwork.ts [hunk 1]
```diff
@@ -1,103 +1,221 @@
...
+        in: R.String.Or(R.Null),
+        cm: R.String.Or(R.Null),
+      }),
+      editions: R.String,
+      display_price_currency: R.String,
+      availability: R.String,
+      availability_hidden: R.Boolean,
+    })
+  ),
+  cultural_makers: R.Array(R.Unknown),
```
Comment (dzucconi) [inline src/gravity/Artwork.ts:113]: I'm taking this to mean that there are no artworks with cultural makers published on staging.

File: src/gravity/Artwork.ts [hunk 2]
```diff
@@ -1,103 +1,221 @@
...
+  sold: R.Boolean,
+  date: R.String,
+  dimensions: R.Record({ in: R.Null.Or(R.String), cm: R.Null.Or(R.String) }),
+  price: R.String,
+  series: R.Null.Or(R.String),
+  availability: R.String,
+  availability_hidden: R.Boolean,
+  ecommerce: R.Boolean,
+  offer: R.Boolean,
+  tags: R.Array(R.Unknown),
```
Comment (dzucconi) [inline src/gravity/Artwork.ts:160]: Similarly.

**Comment by dzucconi** [comment]
OK — so noticing I never received any `cultural_makers` or `tags` previously I realized what happens here is that once we start getting up in number of pages, the offset based pagination starts to fall apart and begins timing out, so we can't actually iterate over everything. So I built the remainder of the representation by sampling artworks from both cultural makers and tags and then iterating over those IDs. Which leaves us with what should be an accurate picture, provided there aren't any other surprises lurking.

File: src/gravity/Artwork.ts [hunk 3]
```diff
@@ -1,103 +1,230 @@
...
+        id: R.String,
+        category_type: R.String,
+        name: R.String,
+        internal: R.Boolean,
+      })
+    ),
+    _id: R.String,
+    id: R.String,
+    default_profile_id: R.Null.Or(R.String),
+    default_profile_public: R.Null.Or(R.Boolean),
```
Comment (dzucconi) [inline src/gravity/Artwork.ts:45]: This is upsetting 😄

File: src/gravity/Artwork.ts [hunk 4]
```diff
@@ -1,103 +1,230 @@
...
-  Record,
-  Static,
-  Undefined,
-  Null,
-} from "runtypes";
-import { EmbeddedArtist } from "./Artist";
-import { EditionSet } from "./EditionSet";
-import { Image } from "./Image";
-import { CulturalMaker } from "./CulturalMaker";
-import { Partner } from "./Partner";
```
Comment (dzucconi) [inline src/gravity/Artwork.ts:15]: Ultimately I think composing types from other places is too error prone. Each type should ultimately represent an individual API response.

**Comment by artsyit** [comment]
:rocket: PR was released in [`v0.1.0`](https://github.com/artsy/quartz/releases/tag/v0.1.0) :rocket:

**Pull request merged**